[scripts-audit] Rollup PR 2021-02-26 - cmake scripts not found #16440

**Describe the bug**
- See PR #16440 vcpkg helper cmake port scripts not found
- Vcpkg appears to not know how to find the new portfile functions `vcpkg_cmake_build` `vcpkg_cmake_configure` `vcpkg_cmake_install` and `vcpkg_cmake_config_fixup` 

**Environment**
- OS: [Microsoft Windows [Version 10.0.19042.844]] 
        (I have not tested this on my Linux OpenSUSE 15.2 on WSL2 and Hyper-V)
- Compiler: 
  - Visual Studio 2017 Edition [Version 15.9.33]
    - Visual Studio Toolset [Version 14.16.27023]
    - Microsoft (R) C/C++ Optimizing Compiler [Version 19.16.27045 for x64]
  - Visual Studio 2019 Edition [Version 16.8.6]
    - Visual Studio Toolset [Version 14.28.29333]
    - Microsoft (R) C/C++ Optimizing Compiler [Version 19.28.29337 for x64]

**To Reproduce**
Steps to reproduce the behavior:
1. System default triplet is `x64-windows`
2. From Powershell Core 7.2 prompt with VS variables set
3.  `git pull`  updates vcpkg instance with PR #16640
4. Update local port(s) portfile.cmake to use new vcpkg CMake port helper functions from #16440 as the following messages are now present:
   - See attached: [New vcpkg CMake Deprecation Warnings.txt](https://github.com/microsoft/vcpkg/files/6058754/New.vcpkg.CMake.Deprecation.Warnings.txt)
5. `./vcpkg install vcpkg-cmake vcpkg-cmake-config --recurse --keep-going` installed successfully for x64-windows
6. `./vcpkg install [port]` - in this case a local copy of rapidstring as vcpkg no longer maintains the port
7. See error - console log only:
>PS D:\vcpkgMgr\2019\test> vcpkg install rapidstring
>Computing installation plan...
>The following packages will be built and installed:
>    rapidstring[core]:x64-windows -> 2018-08-03#1002
>Starting package 1/1: rapidstring:x64-windows
>Building package rapidstring[core]:x64-windows...
>-- Using cached D:/vcpkgMgr/2019/test/downloads/cenit-rapidstring-2a5fa4a42b37b00e1388f4161b1a96876e25b555.tar.gz
>-- Cleaning sources at D:/vcpkgMgr/2019/test/buildtrees/rapidstring/src/876e25b555-4a0d79fa33.clean. Use --editable to skip cleaning for the packages you specify.
>-- Extracting source D:/vcpkgMgr/2019/test/downloads/cenit-rapidstring-2a5fa4a42b37b00e1388f4161b1a96876e25b555.tar.gz
>-- Using source at D:/vcpkgMgr/2019/test/buildtrees/rapidstring/src/876e25b555-4a0d79fa33.clean
>CMake Error at ports/rapidstring/portfile.cmake:11 (vcpkg_cmake_configure):
>  **Unknown CMake command "vcpkg_cmake_configure".**
>Call Stack (most recent call first):
>scripts/ports.cmake:139 (include)
>
>Error: Building package rapidstring:x64-windows failed with: BUILD_FAILED
>Please ensure you're using the latest portfiles with `.\vcpkg update`, then
>submit an issue at https://github.com/Microsoft/vcpkg/issues including:
>  Package: rapidstring:x64-windows
>  Vcpkg version: 2021-01-13-8027bea90b47d50fd6090bbf5ded330f00f87570

**Expected behavior**
vcpkg should install port using new helper cmake port scripts

**Failure logs** 
-None - only console log shown above

**Additional context**
After install the ports vcpkg
- The new scripts are located  at these locations after install:
  - Port vcpkg-cmake-config
    - [vcpkg-root]\installed\x64-windows\share\vcpkg-cmake-config
    - [vcpkg-root]\packages\vcpkg-cmake-config_x64-windows\share\vcpkg-cmake-config
    - [vcpkg-root]\ports\vcpkg-cmake-config
  - Port vcpkg-cmake
    - [vcpkg-root]\installed\x64-windows\share\vcpkg-cmake
    - [vcpkg-root]\packages\vcpkg-cmake-config_x64-windows\share\vcpkg-cmake
    - [vcpkg-root]\ports\vcpkg-cmake
  - Vcpkg List showing x64-windows version of ports installed:
>PS D:\vcpkgMgr\2019\test> vcpkg list
>vcpkg-cmake-config:x64-windows                     2021-02-26
>vcpkg-cmake:x64-windows                            2021-02-26
>vcpkg-gfortran:x64-windows                         3                Metaport to install gfortran dependencies from m...
>vcpkg-gfortran:x86-windows                         3                Metaport to install gfortran dependencies from m...
>yasm-tool-helper:x64-windows                       2020-10-12       This port provides helper functions for ports ne...
>yasm-tool-helper:x86-windows                       2020-10-12       This port provides helper functions for ports ne...
>yasm-tool:x86-windows                              2020-10-12       A modular assembler. This port is intended to bu...

Note 1: Vcpkg.exe was built from the latest clone of microsoft/vcpkg-tool with CMake as of ~ 03/01/2021 01:45 UTC
Note 2: Downloaded [vcpkg.exe](https://github.com/microsoft/vcpkg-tool/releases/download/2021-02-24-d67989bce1043b98092ac45996a8230a059a2d7e/vcpkg.exe) with same results only difference is the sha commit string version Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e 
Note 3: Tried to install ports `vcpkg-cmake:x86-windows` and `vcpkg-cmake-config:x86-windows` in case it is like port `yasm-tool` which is only installed as x86-windows, but was not allowed to install x86-windows triplet with message: 
>PS D:\vcpkgMgr\2019\test> vcpkg install vcpkg-cmake vcpkg-cmake-config --triplet=x86-windows --recurse --keep-going
>Computing installation plan...
>warning: vcpkg appears to be in a Visual Studio prompt targeting x64 but is installing packages for x86-windows. Consider using --triplet x64-windows or --triplet x64-uwp.
>The following packages will be built and installed:
>    vcpkg-cmake[core]:x86-windows -> 2021-02-26
>    vcpkg-cmake-config[core]:x86-windows -> 2021-02-26
>Starting package 1/2: vcpkg-cmake:x86-windows
>Building package vcpkg-cmake[core]:x86-windows...
>CMake Error at ports/vcpkg-cmake/portfile.cmake:2 (message):
>  vcpkg-cmake is a host-only port; please mark it as a host port in your
  dependencies.
>Call Stack (most recent call first):
>  scripts/ports.cmake:139 (include)
>
>Error: Building package vcpkg-cmake:x86-windows failed with: BUILD_FAILED
>Elapsed time for package vcpkg-cmake:x86-windows: 712.7 ms
>Starting package 2/2: vcpkg-cmake-config:x86-windows
>Building package vcpkg-cmake-config[core]:x86-windows...
>CMake Error at ports/vcpkg-cmake-config/portfile.cmake:2 (message):
>  vcpkg-cmake-config is a host-only port; please mark it as a host port in
  your dependencies.
>Call Stack (most recent call first):
>  scripts/ports.cmake:139 (include)
>
>Error: Building package vcpkg-cmake-config:x86-windows failed with: BUILD_FAILED
>Elapsed time for package vcpkg-cmake-config:x86-windows: 32.85 ms

**Attached**
- Portfiles for local copy of `rapidstring`:
  - CMakeLists.txt: [CMakeLists.txt](https://github.com/microsoft/vcpkg/files/6058639/CMakeLists.txt)
  - portfile.cmake: [portfile.cmake.txt](https://github.com/microsoft/vcpkg/files/6058641/portfile.cmake.txt) with updates as follows:
    - vcpkg_install_cmake has been deprecated in favor of vcpkg_cmake_install
    - vcpkg_build_cmake has been deprecated in favor of vcpkg_cmake_build
    - vcpkg_fixup_cmake_targets has been deprecated in favor of  vcpkg_cmake_config_fixup
  - vcpkg.json: [vcpkg.json.txt](https://github.com/microsoft/vcpkg/files/6058642/vcpkg.json.txt)

--- a/docs/maintainers/maintainer-guide.md
+++ b/docs/maintainers/maintainer-guide.md
@@ -43,7 +43,6 @@ At this time, the following helpers are deprecated:
 2. `vcpkg_apply_patches()` should be replaced by the `PATCHES` arguments to the "extract" helpers (e.g. [`vcpkg_from_github()`](vcpkg_from_github.md))
 3. `vcpkg_build_msbuild()` should be replaced by [`vcpkg_install_msbuild()`](vcpkg_install_msbuild.md)
 4. `vcpkg_copy_tool_dependencies()` should be replaced by [`vcpkg_copy_tools()`](vcpkg_copy_tools.md)
-5. `vcpkg_configure_cmake()`, `vcpkg_build_cmake()`, and `vcpkg_install_cmake()` should be replaced by the helper functions in [`vcpkg-cmake`](ports/vcpkg-cmake.md).
 
 ### Avoid excessive comments in portfiles
 

--- a/docs/maintainers/maintainer-guide.md
+++ b/docs/maintainers/maintainer-guide.md
@@ -43,6 +43,7 @@ At this time, the following helpers are deprecated:
 2. `vcpkg_apply_patches()` should be replaced by the `PATCHES` arguments to the "extract" helpers (e.g. [`vcpkg_from_github()`](vcpkg_from_github.md))
 3. `vcpkg_build_msbuild()` should be replaced by [`vcpkg_install_msbuild()`](vcpkg_install_msbuild.md)
 4. `vcpkg_copy_tool_dependencies()` should be replaced by [`vcpkg_copy_tools()`](vcpkg_copy_tools.md)
+5. `vcpkg_configure_cmake()`, `vcpkg_build_cmake()`, and `vcpkg_install_cmake()` should be replaced by the helper functions in [`vcpkg-cmake`](ports/vcpkg-cmake.md).
 
 ### Avoid excessive comments in portfiles
 
@@ -229,7 +230,7 @@ If you're updating multiple ports at the same time, instead you can run:
 vcpkg x-add-version --all
 ```
 
-To update the files for all modified ports at once.  
+To update the files for all modified ports at once.
 
 _NOTE: These commands require you to have committed your changes to the ports before running them. The reason is that the Git SHA of the port directory is required in these version files. But don't worry, the `x-add-version` command will warn you if you have local changes that haven't been committed._
 

--- a/scripts/cmake/vcpkg_build_cmake.cmake
+++ b/scripts/cmake/vcpkg_build_cmake.cmake
@@ -36,14 +36,11 @@ You can use the alias [`vcpkg_install_cmake()`](vcpkg_configure_cmake.md) functi
 
 function(vcpkg_build_cmake)
     cmake_parse_arguments(PARSE_ARGV 0 "arg"
-        "DISABLE_PARALLEL;ADD_BIN_TO_PATH;Z_VCPKG_DISABLE_DEPRECATION_MESSAGE"
+        "DISABLE_PARALLEL;ADD_BIN_TO_PATH"
         "TARGET;LOGFILE_ROOT"
         ""
     )
 
-    if(NOT arg_Z_VCPKG_DISABLE_DEPRECATION_MESSAGE)
-        message(DEPRECATION "vcpkg_build_cmake has been deprecated in favor of vcpkg_cmake_build from the vcpkg-cmake port.")
-    endif()
     if(Z_VCPKG_CMAKE_BUILD_GUARD)
         message(FATAL_ERROR "The ${PORT} port already depends on vcpkg-cmake; using both vcpkg-cmake and vcpkg_build_cmake in the same port is unsupported.")
     endif()

--- a/scripts/cmake/vcpkg_configure_cmake.cmake
+++ b/scripts/cmake/vcpkg_configure_cmake.cmake
@@ -69,7 +69,6 @@ This command supplies many common arguments to CMake. To see the full list, exam
 #]===]
 
 function(vcpkg_configure_cmake)
-    message(DEPRECATION "vcpkg_configure_cmake has been deprecated in favor of vcpkg_cmake_configure from the vcpkg-cmake port.")
     if(Z_VCPKG_CMAKE_CONFIGURE_GUARD)
         message(FATAL_ERROR "The ${PORT} port already depends on vcpkg-cmake; using both vcpkg-cmake and vcpkg_configure_cmake in the same port is unsupported.")
     endif()

--- a/scripts/cmake/vcpkg_fixup_cmake_targets.cmake
+++ b/scripts/cmake/vcpkg_fixup_cmake_targets.cmake
@@ -53,7 +53,6 @@ Replace `${CURRENT_INSTALLED_DIR}` with `${_IMPORT_PREFIX}` in configs and targe
 #]===]
 
 function(vcpkg_fixup_cmake_targets)
-    message(DEPRECATION "vcpkg_fixup_cmake_targets has been deprecated in favor of vcpkg_cmake_config_fixup from the vcpkg-cmake-config port.")
     if(Z_VCPKG_CMAKE_CONFIG_FIXUP_GUARD)
         message(FATAL_ERROR "The ${PORT} port already depends on vcpkg-cmake-config; using both vcpkg-cmake-config and vcpkg_fixup_cmake_targets in the same port is unsupported.")
     endif()

--- a/scripts/cmake/vcpkg_install_cmake.cmake
+++ b/scripts/cmake/vcpkg_install_cmake.cmake
@@ -26,7 +26,6 @@ parameter.
 #]===]
 
 function(vcpkg_install_cmake)
-    message(DEPRECATION "vcpkg_install_cmake has been deprecated in favor of vcpkg_cmake_install from the vcpkg-cmake port.")
     if(Z_VCPKG_CMAKE_INSTALL_GUARD)
         message(FATAL_ERROR "The ${PORT} port already depends on vcpkg-cmake; using both vcpkg-cmake and vcpkg_install_cmake in the same port is unsupported.")
     endif()

